packages/workspace/src/browser/workspace-service.ts

```diff
@@ -24,11 +24,17 @@ export class WorkspaceService {
         (async () => {
             const root = await this.tryRoot;
             if (root) {
-                watcher.watchFileChanges(new URI(root.uri));
+                const uri = new URI(root.uri);
+                this.updateTitle(uri);
+                watcher.watchFileChanges(uri);
             }
         })();
     }
 
+    protected updateTitle(uri: URI): void {
+        document.title = `Theia - ${uri.displayName}`;
```

Comment:
Would it be fine for yangster to have `Theia` in the title?

Reply:
Just because I have verified this with a WS with a `yang` file this does not do anything with Yangster.

Reply:
Generally: if one compose a product he either should be able to customize the title or we could always show `uri.displayName` as a title as VS Code does.

Reply:
I agree, but the PR fixes exactly what is described in the [issues](https://github.com/theia-ide/theia/issues/687). Can you open a follow-up for further enhancements?

> When using multiple tabs with multiple workspaces they are all titled http://localhost:3000

> We should change this to like Theia <Workspacename>

Reply:
I would vote for just displaying uri.displayName

Reply:
Then please update your original request too. It is more convenient to track PRs with their issues. Especially, later. So we can see what the requirement was and what was eventually implemented. Thanks!

Reply:
Note that I used the wording "to like" Theia <workspace> so it was just a suggestion. but I've updated the issue.

Reply:
That is why I extracted it to a method, that can be overridden. But I agree, this would be bad for products.

Reply:
That would be a good idea in general..for another PR